components/elective/ElectiveStudentListCard.tsx

```diff
@@ -0,0 +1,120 @@
+import EnrollmentIndicator from "@/components/elective/EnrollmentIndicator";
+import PersonCard from "@/components/person/PersonCard";
+import cn from "@/utils/helpers/cn";
+import getLocaleName from "@/utils/helpers/getLocaleName";
+import useLocale from "@/utils/helpers/useLocale";
+import { LangCode, StylableFC } from "@/utils/types/common";
+import { ElectiveSubject } from "@/utils/types/elective";
+import { UserRole } from "@/utils/types/person";
+import {
+  AssistChip,
+  ChipSet,
+  DURATION,
+  EASING,
+  MaterialIcon,
+  Search,
+  Text,
+  transition,
+} from "@suankularb-components/react";
+import { motion } from "framer-motion";
+import { useTranslation } from "next-i18next";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { sort } from "radash";
+import { useState } from "react";
+
+/**
+ * A Card that displays a list of all Students enrolled in an Elective Subject,
+ * filterrable by name.
+ *
+ * @param electiveSubject The Elective Subject to display the Students of.
+ */
+const ElectiveStudentListCard: StylableFC<{
+  electiveSubject: ElectiveSubject;
+}> = ({ electiveSubject, style, className }) => {
+  const locale = useLocale();
+  const { locales } = useRouter();
+  const { t } = useTranslation("elective", { keyPrefix: "detail.students" });
+
+  const [query, setQuery] = useState("");
+
+  const classrooms = Object.fromEntries(
+    electiveSubject.applicable_classrooms.map((classroom) => [
+      classroom.id,
+      classroom,
+    ]),
+  );
+  const students = sort(
+    sort(
+      electiveSubject.students.filter((student) =>
+        locales?.some((locale) =>
+          getLocaleName(locale as LangCode, student).includes(query),
+        ),
+      ),
+      (student) => student.classroom?.number || 1000,
+    ),
+    (student) => (student.classroom ? -1 : student.class_no!),
```

Comment:
How does this sorting condition work? I read the document for this, but I still don't understand.

Reply:
I just added a commit with more comments in this section. Sometimes spaghetti code happens and the best thing you can do is add comments explaining what it does.

Check it out, and let me know if you have further questions!